scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala

```diff
@@ -72,7 +81,13 @@ trait LocalSourceOverride extends SchemedSource {
    * @param sinkMode The mode for handling output conflicts.
    * @returns A tap.
    */
-  def createLocalTap(sinkMode: SinkMode): Tap[_, _, _] = new FileTap(localScheme, localPath, sinkMode)
+  def createLocalTap(sinkMode: SinkMode): Tap[JobConf, _, _] = {
+    val taps = localPaths.map {
+      p: String =>
+        CastFileTap(new FileTap(localScheme, p, sinkMode))
+    }.toList
```

Comment:
can we do:
```scala
taps match {
  case Nil => throw new InvalidSourceException("LocalPaths is empty")
  case oneTap :: Nil => oneTap
  case many => new ScaldingMultiSourceTap(many)
}
```
This way, we don't use the multisource path, which is new in this case, unless you need the new functionality, protecting users who don't make any change.